grader/reject.go

```diff
@@ -38,7 +38,17 @@ func (this *RejectWindowMax) String() string {
             nextTime.Format(time.RFC1123), delta.String());
 }
 
-func checkForRejection(assignment *model.Assignment, submissionPath string, user string, message string) (RejectReason, error) {
+type RejectMissingLateAcknowledgment struct {}
+
+func (this *RejectMissingLateAcknowledgment) String() string {
+    return "This assignment is past the due date and the late policy will be applied."
+}
+
+func checkForRejection(assignment *model.Assignment, submissionPath string, user string, message string, lateAcknowledgment bool) (RejectReason, error) {
```

Comment:
I'm not a fan of the specific argument (`lateAcknowledgment`) to bypass one rejection check.
I would love if we could generalize this.

I am specifically thinking about the future when we have direct/proxy submissions (a TA submitting on behalf of a student).
We may want to bypass all rejection checks.

Up in the API endpoint I mentioned using a rejection reason field instead of a specific one for being late.
We can expand on that concept and have a rejection reason bitset enum (like FileMode: https://pkg.go.dev/io/fs#FileMode).
The API endpoint can translate this enum to a string, and here we can pass a single uint32 instead of many booleans (this PR only has one, but this pattern would require more in the future).
Then, each specific rejection check can look at this value and see if their specific rejection reason should be ignored.
Note that this type would probably be in addition to our existing RejectReason type (since that carries more information with it), and not replace it (but you can try if you want).